doc/manual/src/language/identifier.md

```diff
@@ -0,0 +1,40 @@
+# Identifier
+
+An *identifier* is an [ASCII](https://en.wikipedia.org/wiki/ASCII) character sequence that:
+- Starts with a letter (`a-z`, `A-Z`) or underscore (`_`)
+- Can contain any number of:
+  - Letters (`a-z`, `A-Z`)
+  - Digits (`0-9`)
+  - Underscores (`_`)
+  - Apostrophes (`'`)
+  - Hyphens (`-`)
+- Is not one of the [keywords](#keywords)
+
+> **Syntax**
+>
+> *identifier* ~ `[A-Za-z_][A-Za-z0-9_'-]*`
+
+# Keywords
+
+These keywords are reserved and cannot be used as [identifiers](#identifiers):
+
+- [`if`][if]
+- [`then`][if]
+- [`else`][if]
+- [`let`][let]
+- [`in`][let]
+- [`assert`](./syntax.md#assertions)
+- [`inherit`](./syntax.md#inheriting-attributes)
+- [`rec`](./syntax.md#recursive-sets)
+- [`with`](./syntax.md#with-expressions)
+- [`or`](./operators.md#attribute-selection) (see note)
```

Comment:
Why not alphabetize these? I expect that wanting to look up a known keyword in this list is the typical use case, and a seek is better than a scan.

Reply:
The idea was that placing things together that belong together makes it easier on the brain when reading back to back, and scanning 9 items is not a big deal (also there's Ctrl+F if it is).

Reply:
You might be surprised how easy it is to miss the presence of an item in even a small list if that list is not organized the way one would expect. My technical writing instincts generally point towards alphabetizing lists like this unless they're so strongly categorized that they actually belong under separate sub-headers.

But you're the boss. Any list is better than none.